### PR TITLE
qt: Fix OpenWalletActivity::open()

### LINKED ISSUE #18835: debug assert  in walletcontroller.cpp in func getOrCreateWallet, if thread is not gui
hello guys,

code to inspect:
wallet_model->moveToThread(thread());
wallet_model->setParent(this) 

fires on my machine under visual studio 2017, Qt 5.9.7 a debug assert, if thread is from QTimer::singleShot.  if u roll back the code with the slot implementation, then u have no problems. On release it works, but u have a bad feeling.

here is the orginal error msg:
GUI: ASSERT failure in QCoreApplication::sendEvent: "Cannot send events to objects owned by a different thread. Current thread 0x0x351f1ec0. Receiver '' (of type 'WalletController') was created in thread 0x0x21a27c0", N:\Qt\Qtv5.9.7_src\qtbase\src\corelib\kernel\qcoreapplication.cpp, line 563

### PR DESCRIPTION
**Pull request by hebasto**
Fix #18835, I hope :)

### CODE REVIEW

File: src/qt/walletcontroller.cpp [hunk 1]
```diff
@@ -308,6 +308,6 @@ void OpenWalletActivity::open(const std::string& path)
 
         if (wallet) m_wallet_model = m_wallet_controller->getOrCreateWallet(std::move(wallet));
 
-        QTimer::singleShot(0, this, &OpenWalletActivity::finish);
+        QMetaObject::invokeMethod(this, "finish", Qt::QueuedConnection);
```
Comment (promag) [inline src/qt/walletcontroller.cpp:311]: What is the idea? AFAIK `singleShot` does a queued connection too.

Comment (hebasto) [reply src/qt/walletcontroller.cpp:311]: > What is the idea? AFAIK `singleShot` does a queued connection too.

I thought so too. But careful reading https://doc.qt.io/qt-5/qtimer.html didn't point a such statement.

Comment (hebasto) [reply src/qt/walletcontroller.cpp:311]: Also I'm curious why only windows reveals this bug?

**Comment by promag** [comment]
Concept ACK. This will go away once we use latest `invokeMethod` that support pointer to member or lambda.

Looks like `QTimer::singleShot` changed after 5.9.7:
- https://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/kernel/qtimer.cpp?h=v5.9.7#n342
- https://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/kernel/qtimer.cpp#n344

The change in this PR does what latest `singleShot` does so I'd say this should fix the problem. 

I'd improve this PR by
 - assert `invokeMethod` result;
 - do the same in `CreateWalletActivity::createWallet()`.

**Comment by hebasto** [comment]
Closed in favor of #18948.

**Pull request closed without merging**